src/components/custom-input/custom-input.tsx

```diff
@@ -2,12 +2,30 @@ import {TextInput, TextInputProps} from 'react-native';
 import React, {forwardRef} from 'react';
 import * as Styled from './custom-input.styles';
 
-export const CustomInput = forwardRef<TextInput, TextInputProps>(
-  ({...rest}, ref) => {
+interface CustomInputProps extends TextInputProps {
+  errorMessage?: string;
+  onValidation: (e: any) => void;
+  ref: TextInputProps;
+}
+
+export const CustomInput = forwardRef<TextInput, CustomInputProps>(
+  ({onValidation, errorMessage, ...rest}, ref) => {
+    const handlePressOut = e => {
+      onValidation(e);
+    };
+
     return (
-      <Styled.Container>
-        <Styled.Input {...rest} placeholderTextColor={'gray'} ref={ref} />
-      </Styled.Container>
+      <Styled.ContainerCustomInput>
+        <Styled.CustomInput
+          {...rest}
+          placeholderTextColor={'gray'}
+          ref={ref}
+          onBlur={handlePressOut}
+        />
+        {errorMessage && (
+          <Styled.ErrorMessage>{errorMessage}</Styled.ErrorMessage>
+        )}
+      </Styled.ContainerCustomInput>
```

Comment:
vc está puxando todos os componentes com `Styled`, pq n chama direto cada um dos componentes?
```suggestion
      <ContainerCustomInput>
        <CustomInput
          {...rest}
          placeholderTextColor={'gray'}
          ref={ref}
          onBlur={handlePressOut}
        />
        {errorMessage && (
          <ErrorMessage>{errorMessage}</Styled.ErrorMessage>
        )}
      </ContainerCustomInput>
```

Reply:
a gente costuma sempre usar o componente direto, pq tem casos que a gente usa `Compound Components`, isso acabar impactando no  entendimento do código

Reply:
Então, pelo o que eu tava estudando, em aplicações que vão crescendo ou já estão grandes, quanto mais vai caminhando, mais importações vão tendo, e vai ficando inviável puxar um por um, ficando muito grande e de difícil manutenção porque a cada vez que criasse outro componente tinha que lembrar de importar ele, usar no código e aumentar desnecessariamente a parte de importações.
Fora também a parte de diferenciação, quando você tem a parte de styled antes, visualmente você já sabe que é um componente local e quando não tem o prefixo, é um componente externo